internal/proxy/docker/docker.go

```diff
@@ -384,29 +419,46 @@ type patchSet struct {
 
 // ContainerTruststorePatcher provides a Docker API proxy that patches the container truststore while running.
 type ContainerTruststorePatcher struct {
-	Cert        x509.Certificate
-	EnvVars     []string
-	JavaEnvVar  bool
-	proxySocket string
-	patchMap    map[string]*patchSet
-	m           sync.Mutex
-	created     atomic.Uint32
+	cert            x509.Certificate
+	envVars         []string
+	javaEnvVar      bool
+	requiredNetwork string // TODO: Not a good fit for this abstraction
```

Comment:
nit: is requiredNetwork the right name for this? Required makes it sound like we're checking if it's already present, wdyt of patchedNetwork, or networkOverride, or maybe just ContainerTruststorePatcher.network?

Reply:
i didn't want patched because it's not unpatched. "network" is tricky because it's not clear from the name of the abstraction how a "network" would relate to it. Override is closer to what I was aiming for and i do think it's a better option than required. Done.